x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/streams/group_stream.ts

```diff
@@ -52,40 +52,42 @@ export class GroupStream extends StreamActiveRecord<Streams.GroupStream.Definiti
       }
     }
 
-    const cascadingChanges: StreamChange[] = [];
-    if (missingMembers.length > 0) {
-      for (const member of missingMembers) {
+    const existsAsDataStream = await Promise.all(
+      missingMembers.map(async (member) => {
         try {
           const dataStreamResult =
             await this.dependencies.scopedClusterClient.asCurrentUser.indices.getDataStream({
               name: member,
             });
 
-          if (dataStreamResult.data_streams.length > 0) {
-            cascadingChanges.push({
-              type: 'upsert',
-              definition: {
-                name: member,
-                description: '',
-                ingest: {
-                  classic: {},
-                  lifecycle: {
-                    inherit: {},
-                  },
-                  processing: {
-                    steps: [],
-                  },
-                },
-              },
-            });
-          }
+          return dataStreamResult.data_streams.length > 0 ? member : null;
         } catch (error) {
           if (!isNotFoundError(error)) {
             throw error;
           }
+          return null;
         }
-      }
-    }
+      })
+    );
+
+    const cascadingChanges: StreamChange[] = existsAsDataStream
+      .filter((member): member is string => member !== null)
```

Comment:
didn't run but isn't this the wrong way around? It would attempt to upsert all streams that do exist, right?

Reply:
Right... I got confused just skimming it, but that makes sense, LGTM